### PR TITLE
47 categories

### PR DESCRIPTION
**Pull request by Superdoude1999**
Added the category attribute to items, made it so they can only have one category cause in the end it seems more logical. Also modified the editItem and newItem fragments to implement the categories. This branch also contains the searchFragment which isn't completed but forgot to make another branch for it sorry. Addresses issue #47

### CODE REVIEW

File: app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt [hunk 1]
```diff
@@ -25,14 +27,42 @@ class EditItemFragmentTest {
...
     @Test
     fun anItemCanBeEditedAndSeenOnItemsListFragment() {
         onView(withId(R.id.newItemButton)).perform(click())
         onView(withId(R.id.newItemPrompt))
             .check(matches(withText("New Item")))
         onView(withId(R.id.editItemTitle)).perform(
-            typeText(item),
-            closeSoftKeyboard()
+                typeText(item),
+                closeSoftKeyboard()
```
Comment (bafavre) [inline app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt:65]: Why the tab?

File: app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt [hunk 2]
```diff
@@ -49,8 +79,8 @@ class EditItemFragmentTest {
             .check(matches(withText("Edit Item")))
 
         onView(withId(R.id.editItemTitle)).perform(
-            typeText(editedItem),
-            closeSoftKeyboard()
+                typeText(editedItem),
+                closeSoftKeyboard()
```
Comment (bafavre) [inline app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt:83]: Same thing here

File: app/src/main/java/com/example/sharingang/SearchFragment.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+
+class SearchFragment : Fragment() {
+
+    private val viewModel : ItemsViewModel by activityViewModels();
+    private val adapter = ItemsAdapter(ItemListener {  });
+//    private val repo = FirestoreItemRepository()
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/SearchFragment.kt:30]: Don't keep commented code

File: app/src/main/java/com/example/sharingang/SearchFragment.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    private fun searchList(search_name : String?){
+        Log.d("info","Seach function called");
+        if(search_name == null){Log.d("info", "String search was null");}else{
+            Log.d("info", search_name);
+        }
+//        val listItems = repo.getAllItemsLiveData()
+
+
+//        adapter.submitList(searchResults);
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/SearchFragment.kt:54]: Same goes here

**Review by bafavre — COMMENTED** [review]
Nice job!

**Review by v1ruso — CHANGES_REQUESTED** [review]
Just a reminder to merge before, so that you can finish your feature

File: app/src/main/java/com/example/sharingang/NewItemFragment.kt [hunk 5]
```diff
@@ -23,12 +24,27 @@ class NewItemFragment : Fragment() {
...
+                position: Int,
+                id: Long
+            ) {
+                if(adapterView != null){category = position}
+            }
 
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
+            }
+        }
```
Comment (betrisey) [inline app/src/main/java/com/example/sharingang/NewItemFragment.kt]: I think it was better how you did in EditItemFragment. Is it possible to do the same here? (remove this and use `binding.categorySpinner.selectedItemPosition`)

File: app/src/main/java/com/example/sharingang/SearchFragment.kt [hunk 6]
```diff
@@ -0,0 +1,57 @@
+package com.example.sharingang
```
Comment (betrisey) [inline app/src/main/java/com/example/sharingang/SearchFragment.kt:1]: We'll review this later when it's finished

**Review by betrisey — CHANGES_REQUESTED** [review]
Nice work :)
Can you just change the NewItemFragment? (and maybe layout if you have time)
I think it's not a problem if we merge now the SearchFragment even though it's not finished, it doesn't impact the rest of the app.

File: app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt (hunk 1 above)
Comment (betrisey) [reply app/src/androidTest/java/com/example/sharingang/EditItemFragmentTest.kt:65]: I also find it weird but Android Studio is using by default 8 spaces (why so many wtf) when we split a statement in multiple lines
Screenshot 2021-03-18 at 16 27 43

File: app/src/main/res/layout/fragment_search.xml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            android:layout_width="414dp"
+            android:layout_height="659dp"
+            android:scrollbars="vertical"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintVertical_bias="1.0" />
```
[the remark this answers is not in the document]
Comment (Superdoude1999) [reply app/src/main/res/layout/fragment_search.xml:47]: oops used constraint layout but I'll remove that so it doesn't keep dimensions thanks

**Review by betrisey — APPROVED** [review]

File: app/src/main/java/com/example/sharingang/NewItemFragment.kt (hunk 5 above)
Comment (Superdoude1999) [reply app/src/main/java/com/example/sharingang/NewItemFragment.kt]: and yup I did that

**Review by bafavre — APPROVED** [review]

File: app/src/main/java/com/example/sharingang/NewItemFragment.kt [hunk 8]
```diff
@@ -1,39 +1,44 @@
...
+import androidx.navigation.findNavController
+import com.example.sharingang.databinding.FragmentNewItemBinding
+import com.example.sharingang.items.Item
+import com.example.sharingang.items.ItemsViewModel
+
+class NewItemFragment : Fragment() {
+
+    private val viewModel: ItemsViewModel by activityViewModels()
+    var category : Int = 0
+
```
Comment (v1ruso) [inline app/src/main/java/com/example/sharingang/NewItemFragment.kt:21]: Does this variable need to be not private?

File: app/src/main/java/com/example/sharingang/SearchFragment.kt (hunk 6 above)
Comment (v1ruso) [reply app/src/main/java/com/example/sharingang/SearchFragment.kt:1]: Remove it from the branch then?

File: app/src/main/res/layout/fragment_items_list.xml [hunk 9]
```diff
@@ -23,6 +23,12 @@
             android:layout_height="wrap_content"
             android:text="@string/go_to_map" />
 
+        <Button
+            android:id="@+id/gotoSearchButton"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="Search" />
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_items_list.xml:30]: Hardcoded string, add it to string.xml

File: app/src/main/res/layout/fragment_search.xml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        tools:context=".SearchFragment">
+
+        <Button
+            android:id="@+id/sflSearchButton"
+            android:layout_width="104dp"
+            android:layout_height="52dp"
+            android:text="Go"
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:14]: Hardcoded string again

File: app/src/main/res/layout/fragment_search.xml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.5"
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:17]: I think it is safe to just remove the bias

File: app/src/main/res/layout/fragment_search.xml [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+            app:layout_constraintStart_toEndOf="@+id/sflSearchText"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintVertical_bias="0.022" />
+
+        <EditText
+            android:id="@+id/sflSearchText"
+            android:layout_width="231dp"
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:24]: Hardcoded width and height, consider "match_parent"

File: app/src/main/res/layout/fragment_search.xml [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+            android:layout_height="50dp"
+            android:ems="10"
+            android:hint="Search..."
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:27]: Hardcoded string

File: app/src/main/res/layout/fragment_search.xml [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+            android:inputType="textPersonName"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toStartOf="@+id/sflSearchButton"
+            app:layout_constraintHorizontal_bias="0.5"
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:31]: You can remove the bias, I think

File: app/src/main/res/layout/fragment_search.xml [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/itemList"
+            android:layout_width="414dp"
+            android:layout_height="659dp"
+            android:scrollbars="vertical"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintHorizontal_bias="0.0"
```
Comment (v1ruso) [inline app/src/main/res/layout/fragment_search.xml:44]: Same

**Review by v1ruso — APPROVED** [review]

**Pull request merged**